spd/run_spd.py

```diff
@@ -112,6 +113,55 @@ def local_log(data: dict[str, Any], step: int, out_dir: Path) -> None:
         f.write(json.dumps({"step": step, **metrics_without_images}) + "\n")
 
 
+def get_grad_norms_log(
```

Comment:
I think I'd put this outside of run_spd.py. Maybe in one of the utils :)? Not an extremely strong preference, it's just that it's kind of nice that run_spd is focused on the core loop